### ISSUE TITLE
Time change after DST end caused cron failure.

### ISSUE THREAD

**Issue by InterstellarStella** [getsentry/sentry#79813]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

Have a cron monitor scheduled for every day at a specific time of the day.

### Expected Result

After the DST end, the cron should continue working normally.

### Actual Result

After the DST end, and the subsequent time change, the cron started throwing failures every one minute, even though it is scheduled for only once a dat. This started on October 27 at 3:20AM CET, so soon after the time change.
![Image](https://github.com/user-attachments/assets/cde5f0bd-b4bc-4d63-a62e-069260cdcbff)

Example monitors in the [shadow ticket](https://getsentry.atlassian.net/browse/FEEDBACK-2329).

### Product Area

Crons

┆Issue is synchronized with this [Jira Improvement](https://getsentry.atlassian.net/browse/FEEDBACK-2329) by [Unito](https://www.unito.io)

**Comment by evanpurkhiser** [comment]
Duplicate of https://github.com/getsentry/sentry/issues/66868
